### PR TITLE
Added the RBAC for druid leader election.

### PR DESCRIPTION
**Pull request by abdasgupta**
It is required for https://github.com/gardener/etcd-druid/pull/133

### CODE REVIEW

**Comment by rfranzke** [comment]
Please fix the unit tests.
/author-action

File: pkg/operation/botanist/component/etcd/bootstrap_test.go [hunk 1]
```diff
@@ -153,6 +153,17 @@ rules:
   - update
   - patch
   - create
+- apiGroups:
+  - coordination.k8s.io
+  resourceNames:
+  - druid-leader-election
```
Comment (rfranzke) [inline pkg/operation/botanist/component/etcd/bootstrap_test.go:165]: Is this correct? `druid-leader-election` instead of `etcd-druid-leader-election`? :)

Comment (abdasgupta) [reply pkg/operation/botanist/component/etcd/bootstrap_test.go:165]: I am not sure about that actually. @amshuman-kr tried this resource name to test my PR in druid and he was able to run it. But I have to check if any resource name exists by this name. Wl let you know and update the PR.

Comment (abdasgupta) [reply pkg/operation/botanist/component/etcd/bootstrap_test.go:165]: Okay. I removed the resource name and update the PR. It was adding unnecessary complication.

File: pkg/operation/botanist/component/etcd/bootstrap_test.go [hunk 2]
```diff
@@ -420,7 +429,6 @@ status: {}
 				gomock.InOrder(
 					c.EXPECT().Get(ctx, kutil.Key(namespace, managedResourceSecretName), gomock.AssignableToTypeOf(&corev1.Secret{})),
 					c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&corev1.Secret{})).Do(func(ctx context.Context, obj client.Object, opts ...client.UpdateOption) {
-						Expect(obj).To(DeepEqual(managedResourceSecret))
```
Comment (vpnachev) [inline pkg/operation/botanist/component/etcd/bootstrap_test.go]: why is this line removed?

File: pkg/operation/botanist/component/etcd/bootstrap_test.go (hunk 1 above)
Comment (vpnachev) [reply pkg/operation/botanist/component/etcd/bootstrap_test.go:165]: Seems to have been correct https://github.com/gardener/etcd-druid/blob/841f2d016848d24c71423088ba3503f34bb9a3fd/main.go#L58

File: pkg/operation/botanist/component/etcd/bootstrap.go [hunk 3]
```diff
@@ -143,6 +144,11 @@ func (b *bootstrapper) Deploy(ctx context.Context) error {
 					Resources: []string{"etcds/status", "etcds/finalizers"},
 					Verbs:     []string{"get", "update", "patch", "create"},
 				},
+				{
+					APIGroups: []string{coordinationv1.GroupName},
+					Resources: []string{"leases"},
```
Comment (amshuman-kr) [inline pkg/operation/botanist/component/etcd/bootstrap.go:149]: Would it be better to include the `ResourceNames` to avoid giving too much permission via a `ClusterRole`?

File: pkg/operation/botanist/component/etcd/bootstrap_test.go (hunk 2 above)
Comment (amshuman-kr) [reply pkg/operation/botanist/component/etcd/bootstrap_test.go]: I am also not sure why this check is removed. Isn't it needed?

**Review by amshuman-kr — COMMENTED** [review]
I tested the change. They seem to work fine. A couple of questions below.

File: pkg/operation/botanist/component/etcd/bootstrap_test.go (hunk 2 above)
Comment (abdasgupta) [reply pkg/operation/botanist/component/etcd/bootstrap_test.go]: Sorry, oversight. I added back in the updated PR.

**Review by rfranzke — DISMISSED** [review]
/lgtm

**Comment by ialidzhikov** [comment]
Shouldn't this PR be considered only with the new version of etcd-druid?

**Comment by rfranzke** [comment]
Probably, but it doesn't harm to merge this already now, I assume? Shall we go ahead?

**Comment by ialidzhikov** [comment]
I would say “no”. The referenced PR is not even merged. It could be the case that the RBAC needs adaptations.
I will mark the PR as draft. Please make it ready for review when there is a new etcd-druid release.

**Review by amshuman-kr — CHANGES_REQUESTED** [review]
@abdasgupta Can you please also add the permission to `list` `persistentvolumeclaims` to address the following?

```
E0416 16:26:22.457326       1 reflector.go:127] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:225: Failed to watch *v1.PersistentVolumeClaim: failed to list *v1.PersistentVolumeClaim: persistentvolumeclaims is forbidden: User "system:serviceaccount:garden:etcd-druid" cannot list resource "persistentvolumeclaims" in API group "" at the cluster scope
```

**Comment by abdasgupta** [comment]
Hi, the PR https://github.com/gardener/etcd-druid/pull/133 is merged. Would you merge this PR as well?

Thanks

**Comment by abdasgupta** [comment]
Hi, Can you merge this PR as https://github.com/gardener/etcd-druid/pull/133 is merged. @amshuman-kr I made the requested changes

**Comment by ialidzhikov** [comment]
> Hi, Can you merge this PR as [gardener/etcd-druid#133](https://github.com/gardener/etcd-druid/pull/133) is merged. @amshuman-kr I made the requested changes

@abdasgupta , could you please explain me exactly which part of https://github.com/gardener/gardener/pull/3828#issuecomment-815031750 (`Please make it ready for review when there is a new etcd-druid release.`) is not clear?

**Comment by abdasgupta** [comment]
I must have given more importance to this part `The referenced PR is not even merged`. Anyway, sorry that I bothered too early. A new release of ETCD Druid is shortly coming. Would let you know. Thanks.

**Review by rfranzke — APPROVED** [review]
/lgtm
/invite @amshuman-kr @shreyas-s-rao 
/ready

**Comment by abdasgupta** [comment]
New ETCD druid release https://github.com/gardener/etcd-druid/tree/v0.5.0 v0.5.0 is available with the related Druid PR merged in it. We are okay to go ahead with this PR. Thanks.
